test/sql/alter/rename_table/test_rename_table_collision.test

```diff
@@ -0,0 +1,58 @@
+# name: test/sql/alter/rename_table/test_rename_table_collision.test
+# description: Test RENAME TABLE multiple transactions with rename after drop
+# group: [rename_table]
```

Comment:
I would like to see a test involving a chain of renames in the same transaction, e.g.:

```sql
CREATE TABLE entry(i INTEGER);

BEGIN TRANSACTION;
-- entry -> entry2
ALTER TABLE entry RENAME TO entry2;

-- entry2 -> entry3
-- entry -> entry2
CREATE TABLE entry(i INTEGER);
ALTER TABLE entry2 RENAME TO entry3;
ALTER TABLE entry RENAME TO entry2;

-- entry3 -> entry4
-- entry2 -> entry3
-- entry -> entry2
CREATE TABLE entry(i INTEGER);
ALTER TABLE entry3 RENAME TO entry4;
ALTER TABLE entry2 RENAME TO entry3;
ALTER TABLE entry RENAME TO entry2;

-- etc...
```

Now if the transaction rolls back, "entry" needs to go back to pointing to the initial table, and subsequent transactions need to be able to create entry2/entry3/entry4/etc again. If the transaction commits, `entry2`, `entry3` and `entry4` need to be accessible to other transactions and `entry` needs to be gone.